apps/dashboard/src/components/contract-components/contract-deploy-form/custom-contract.tsx

```diff
@@ -837,23 +841,36 @@ export const CustomContractForm: React.FC<CustomContractFormProps> = ({
               {metadata?.deployType === "standard" && (
                 <>
                   {/* Deterministic deploy */}
-                  <CheckboxWithLabel>
-                    <Checkbox
-                      {...form.register("deployDeterministic")}
-                      checked={form.watch("deployDeterministic")}
-                      onCheckedChange={(c) =>
-                        form.setValue("deployDeterministic", !!c)
-                      }
-                    />
-                    <ToolTipLabel label="Allows having the same contract address on multiple chains. You can control the address by specifying a salt for create2 deployment below">
-                      <div className="inline-flex gap-1.5 items-center">
-                        <span className="tex-sm">
-                          Deploy at a deterministic address
-                        </span>
-                        <FiHelpCircle className="size-4" />
-                      </div>
-                    </ToolTipLabel>
-                  </CheckboxWithLabel>
+
+                  <div className="flex flex-col gap-3">
+                    <CheckboxWithLabel>
+                      <Checkbox
+                        {...form.register("deployDeterministic")}
+                        checked={form.watch("deployDeterministic")}
+                        onCheckedChange={(c) =>
+                          form.setValue("deployDeterministic", !!c)
+                        }
+                      />
+                      <ToolTipLabel label="Allows having the same contract address on multiple chains. You can control the address by specifying a salt for create2 deployment below">
+                        <div className="inline-flex gap-1.5 items-center">
+                          <span className="tex-sm">
+                            Deploy at a deterministic address
+                          </span>
+                          <FiHelpCircle className="size-4" />
+                        </div>
+                      </ToolTipLabel>
+                    </CheckboxWithLabel>
+
+                    {shouldShowDeterministicDeployWarning && (
+                      <Alert variant="warning">
+                        <CircleAlertIcon className="size-5" />
+                        <AlertTitle>
+                          Deterministic deployment will only work if you use the
```

Comment:
instead of 'will only work if...' would probably say 'would only result in the same contract address if...'

Reply:
Updated 👍